.github/workflows/haskell-ci.yml

```diff
@@ -0,0 +1,66 @@
+name: Tests
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: CI
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        resolver:
+          [
+            "nightly-2023-05-15",
+            "lts-20.21",
+            "lts-19.33",
+            "lts-18.28",
+          ]
+        include:
+          - resolver: "nightly-2023-05-15"
+            os: ubuntu-latest
+            ghc: "9.4.5"
+            cabal: latest
+            stack: latest
+          - resolver: "lts-20.21"
+            os: ubuntu-latest
+            ghc: "9.2.7"
+            cabal: latest
+            stack: latest
+          - resolver: "lts-19.33"
+            os: ubuntu-latest
+            ghc: "9.0.2"
+            cabal: latest
+            stack: latest
+          - resolver: "lts-18.28"
+            os: ubuntu-latest
+            ghc: "8.10.7"
+            cabal: latest
+            stack: latest
+
+    steps:
+      - name: Setup GHC
+        uses: haskell/actions/setup@v2
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          enable-stack: true
+          stack-version: ${{ matrix.stack }}
+          cabal-version: ${{ matrix.cabal }}
```

Comment:
I don't think you need to install cabal if you're only using stack. I'd remove this line from here + all "cabal: latest" from the matrix configuration.